lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyDiagnosticParticipant.java

```diff
@@ -261,4 +300,214 @@ private void validateConfigElements(DOMDocument domDocument, List<Diagnostic> di
+
+        for (DOMNode featureNode : features) {
+            DOMNode featureTextNode = (DOMNode) featureNode.getChildNodes().item(0);
+            if (LibertyConstants.FEATURE_ELEMENT.equals(featureNode.getLocalName())
+                    && featureTextNode != null && featureTextNode.getTextContent() != null) {
+                String featureName = featureTextNode.getTextContent().trim();
+                if (versionlessFeatures.contains(featureName)) {
+                    // versionless feature
+                    validateVersionLessFeatures(domDocument, list, preferredPlatforms, versionedFeatures, featureTextNode, featureName);
+                }
+            }
+
+        }
+    }
+
+    /**
+     * check for versionless feature version identification
+     *      1) if versionless feature is specified and no versioned feature or platform is specified, throw error
+     *      2) if versionless feature is specified and no platform is specified and atleast one versioned feature is specified
+     *          if no common platform is found for all versioned features, throw error
+     *          if there are more than one common platform, throw error
+     *          if common platform is found, then
+     *              find all required and required tolerates features for versionless feature
+     *              find all platforms for above features
+     *              check common platforms is present in feature platforms
+     *              throw error if not found
+     *
+     *      3) if versionless feature is specified and platform(s) is specified
+     *          find all required and required tolerates features for versionless feature
+     *          find all platforms for above features
+     *          check common platforms is present in feature platforms
+     *          throw error if not found
+     *
+     * @param domDocument
+     * @param list
+     * @param preferredPlatforms
+     * @param versionedFeatures
+     * @param versionLessFeatureTextNode
+     * @param featureName
+     */
+    private void validateVersionLessFeatures(DOMDocument domDocument, List<Diagnostic> list, Set<String> preferredPlatforms, Set<String> versionedFeatures,DOMNode versionLessFeatureTextNode, String featureName) {
+        LibertyRuntime runtimeInfo = LibertyUtils.getLibertyRuntimeInfo(domDocument);
+        String libertyVersion = runtimeInfo == null ? null : runtimeInfo.getRuntimeVersion();
+        String libertyRuntime = runtimeInfo == null ? null : runtimeInfo.getRuntimeType();
+
+        final int requestDelay = SettingsService.getInstance().getRequestDelay();
+        if (versionedFeatures.isEmpty() && preferredPlatforms.isEmpty()) {
+            Range range = XMLPositionUtility.createRange(versionLessFeatureTextNode.getStart(),
+                    versionLessFeatureTextNode.getEnd(), domDocument);
+            String message = "ERROR: Need to specify any platform or at least one versioned feature.";
```

Comment:
This is the message to change:

"ERROR: The " +featureName+" versionless feature cannot be resolved. Specify a platform or a feature with a version to enable resolution."

Reply:
Ok this message is corrected